src/panels/config/integrations/ha-config-integration-page.ts

```diff
@@ -503,16 +503,27 @@ class HaConfigIntegrationPage extends SubscribeMixin(LitElement) {
               <mwc-list>
                 ${normalEntries.map((item) => this._renderConfigEntry(item))}
               </mwc-list>
+
               <div class="card-actions">
-                <ha-button @click=${this._addIntegration}>
-                  ${this._manifest?.integration_type
-                    ? this.hass.localize(
-                        `ui.panel.config.integrations.integration_page.add_${this._manifest.integration_type}`
-                      )
-                    : this.hass.localize(
-                        `ui.panel.config.integrations.integration_page.add_entry`
+                ${this._manifest?.config_flow ||
+                (this._manifest?.integration_type &&
+                  this._manifest?.integration_type?.toString() === "virtual")
```

Comment:
Instead fix the typing
```suggestion
                  this._manifest?.integration_type === "virtual")
```

Reply:
Some variant of that don’t work for me but I’ll give it a try again. Thanks for the feedback.

Reply:
You should add `virtual` to `IntegrationType`